Source/FieldSolver/SpectralSolver/SpectralFieldDataRZ.cpp

```diff
@@ -506,8 +505,8 @@ SpectralFieldDataRZ::ForwardTransform (const int lev,
         field_mf_t_copy[mfi].copy<amrex::RunOn::Device>(field_mf_t[mfi], 0, 0, 1); // Real part of mode 0
         field_mf_r_copy[mfi].setVal<amrex::RunOn::Device>(0._rt, realspace_bx, 1, 1); // Imaginary part of mode 0
         field_mf_t_copy[mfi].setVal<amrex::RunOn::Device>(0._rt, realspace_bx, 1, 1); // Imaginary part of mode 0
-        field_mf_r_copy[mfi].copy<amrex::RunOn::Device>(field_mf_r[mfi], 1, 2, 2*n_rz_azimuthal_modes-2);
-        field_mf_t_copy[mfi].copy<amrex::RunOn::Device>(field_mf_t[mfi], 1, 2, 2*n_rz_azimuthal_modes-2);
+        field_mf_r_copy[mfi].copy<amrex::RunOn::Device>(field_mf_r[mfi], 1, 2, 2*n_rz_azimuthal_modes-2); // why -2 not -1 or 0?
+        field_mf_t_copy[mfi].copy<amrex::RunOn::Device>(field_mf_t[mfi], 1, 2, 2*n_rz_azimuthal_modes-2); // why -2 not -1 or 0?
```

Comment:
The multifab `field_mf_r_copy` includes the imaginary part of mode 0 (even though is it all zeros) and so has a total of `2*n_rz_azimuthal_modes` components. The lines above handle mode 0. These lines handle the rest of the modes, so the number of components copied is `2*(n_rz_azimuthal_modes - 1)`.